ui/app/src/components/GitManagement/PublishCommitDialog/PublishCommitDialog.tsx

```diff
@@ -145,7 +150,10 @@ export function PublishCommitDialog(props: PublishCommitDialogProps) {
             <PublishOnDemandForm
               mode="git"
               formData={data}
-              setFormData={(newData) => setState(newData)}
+              setFormData={(newData) => {
+                fnRefs.current.onSubmittingAndOrPendingChange({ hasPendingChanges: true });
```

Comment:
I don't think this covers blanking the fields? (e.g. backspace)
Also, as soon as the dialog is opened, it has the commit id — the whole reason of this ticket — which should disallow closing. 

Make sure you're testing in Studio, not just on the isolated dev server app.

Reply:
I had a misunderstanding regarding the whole pre-filled id thing, I updated so now it won't close with the pre-filled commit id.